server/controller/src/main/java/ai/starwhale/mlops/domain/job/converter/JobBoConverter.java

```diff
@@ -169,19 +171,30 @@ public Job fromEntity(JobEntity jobEntity) {
 
     public Job fillStepsAndTasks(Job job) {
         List<StepEntity> stepEntities = stepMapper.findByJobId(job.getId());
-        List<Step> steps = stepEntities.parallelStream().map(stepConverter::fromEntity)
-                .peek(step -> {
-                    step.setJob(job);
-                    List<TaskEntity> taskEntities = taskMapper.findByStepId(step.getId());
-                    List<Task> tasks = taskBoConverter.fromTaskEntity(taskEntities, step);
-                    step.setTasks(tasks);
-                    if (step.getStatus() == StepStatus.RUNNING) {
-                        if (job.getCurrentStep() != null) {
-                            log.error("ERROR!!!!! A job has two running steps job id: {}", job.getId());
-                        }
-                        job.setCurrentStep(step);
-                    }
-                }).collect(Collectors.toList());
+        List<Step> steps = stepEntities.stream().map(entity -> {
+            try {
+                var step = stepConverter.fromEntity(entity);
+                if (step.getResourcePool() == null) {
+                    // backward compatibility
```

Comment:
Is there a need for backward compatibility? Not like datasets and models, jobs have very shot life cycles. Could we assume that when we upgrade controller there is no running jobs?

Reply:
This func is used by `JobBoConverter::fromEntity` and the `fromEntity` func is used in multiple places